masu/database/sql/reporting_ocpinfrastructure_provider_map.sql

```diff
@@ -0,0 +1,78 @@
+CREATE TEMPORARY TABLE ocp_infrastructure_temp AS (
+    SELECT aws.cost_entry_bill_id,
+        ocp.cluster_id
+    FROM (
+        SELECT aws.*,
+            key,
+            value
+        FROM reporting_awscostentrylineitem_daily as aws,
+            jsonb_each_text(aws.tags) labels
+    ) aws
+    JOIN reporting_ocpusagelineitem_daily as ocp
+        ON aws.resource_id = ocp.resource_id
+            OR (aws.key = 'openshift_project' AND aws.value = ocp.namespace)
+            OR (aws.key = 'openshift_node' AND aws.value = ocp.node)
+            OR (aws.key = 'openshift_cluster' AND aws.value = ocp.cluster_id)
+            OR (aws.key = 'openshift_cluster' AND aws.value = ocp.cluster_alias)
+    WHERE date(aws.usage_start) >= '{start_date}'
+        AND date(aws.usage_start) <= '{end_date}'
+    GROUP BY aws.cost_entry_bill_id, ocp.cluster_id
+)
+;
+
+CREATE TEMPORARY TABLE aws_infrastructure_uuid_temp AS (
+    SELECT provider.uuid as aws_uuid,
+        aws_id_block.cost_bill_id as cost_entry_bill_id,
+        aws_id_block.cluster_id as cluster_id
+    FROM (
+        SELECT aws.id as aws_id,
+            aws.cost_entry_bill_id as cost_bill_id,
+            aws.cluster_id as cluster_id
+        FROM (
+            SELECT awsbill.provider_id as id,
+                ocp_infra_temp.cost_entry_bill_id as cost_entry_bill_id,
+                ocp_infra_temp.cluster_id as cluster_id
+            FROM reporting_awscostentrybill as awsbill
+            LEFT JOIN ocp_infrastructure_temp as ocp_infra_temp
+                ON awsbill.id = ocp_infra_temp.cost_entry_bill_id
+            WHERE awsbill.id = ocp_infra_temp.cost_entry_bill_id
+        ) aws
+    ) aws_id_block
+    LEFT JOIN public.api_provider as provider
+        ON provider.id = aws_id_block.aws_id
+        WHERE provider.id = aws_id_block.aws_id
+)
+;
+
+CREATE TEMPORARY TABLE ocp_uuid_temp AS (
```

Comment:
Similar situation here without sub-queries this would look like 

```
SELECT provider.uuid as ocp_uuid,
    ocp_infra_temp.cost_entry_bill_id,
    ocp_infra_temp.cluster_id
FROM public.api_providerauthentication as authentication
JOIN public.api_provider as provider
    ON provider.authentication_id = authentication.id
JOIN ocp_infrastructure_temp as ocp_infra_temp
    ON authentication.provider_resource_name = ocp_infra_temp.cluster_id
```